protobuf/v1/replica.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "common.proto";
+
+package mayastor.v1;
+
+service ReplicaRpc {
+  // Replica related methods.
+  //
+  // Replica allocates space from storage pool.
+  rpc CreateReplica (CreateReplicaRequest) returns (Replica) {}
+  rpc DestroyReplica (DestroyReplicaRequest) returns (google.protobuf.Empty) {}
+  rpc ListReplicas (ListReplicasOptions) returns (ListReplicasResponse) {}
+  rpc ShareReplica (ShareReplicaRequest) returns (ShareReplicaResponse) {}
+}
+
+message CreateReplicaRequest {
+  string name = 1;  // name of the replica
+  string uuid = 2;  // uuid of the replica
+  string pool = 3;  // name of the pool
+  PoolType pooltype = 4; // type of the pool
```

Comment:
should this be pool uuid instead?
why do we need to specify the type if we're specifying the pool already?

Reply:
Having the pool type here reduces the complexity. 

Consider the case where pooltype is not a field here and we have some Lvs pools and some Lvm Pools.

So first we will have to find the corresponding pool, which involves searching for Lvs pools for the name, then in the LVm pools. once we get the pool, we then proceed with the creation of a volume on that pool.

If pool type is specified, we just search that particular pooltype only. eg: pooltype: LVM, we search for pool "pool0" in LVM, and if not found we just return that pool not found.

Pooltype can be removed from here, just that it adds more complexity.

Reply:
> should this be pool uuid instead?

yep. That can be done.

Reply:
> 
Having the type here may reduce the complexity for the server side but it increases it for the client side, as the client will might have to make that same query right? IMHO I'd prefer to make it simpler for the client

Reply:
Will update it. Replica wont be having a pooltype, but just the pool name and uuid